ui-core/src/components/inputs/TolerancePicker.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useState, useRef, useMemo, useEffect } from 'react';
+import Input, { InputProps } from './Input';
+import Modal from '../Modal';
+import cx from 'classnames';
+import { StatusWithMessage } from './StatusMessage';
+
+type ToleranceRange = { label: string; value: number };
+
+export type ToleranceValues = {
+  minusTolerance: number;
+  plusTolerance: number;
+};
+
+export type TolerancePickerProps = Omit<InputProps, 'value'> & {
+  onToleranceChange: (toleranceValues: ToleranceValues) => void;
+  toleranceRanges: ToleranceRange[];
+  toleranceValues?: ToleranceValues;
+};
+
+const TolerancePicker: React.FC<TolerancePickerProps> = ({
+  onToleranceChange,
+  toleranceRanges,
+  toleranceValues: { minusTolerance, plusTolerance } = {
+    minusTolerance: toleranceRanges[0].value,
+    plusTolerance: toleranceRanges[0].value,
+  },
+  ...inputProps
+}) => {
+  const formatToleranceValue = (minusIndex: number, plusIndex: number) =>
+    `-${toleranceRanges[minusIndex]?.label || ''}/+${toleranceRanges[plusIndex]?.label || ''}`;
+
+  const [showPicker, setShowPicker] = useState(false);
+  const [inputValue, setInputValue] = useState(formatToleranceValue(0, 0));
+  const [warningStatus, setWarningStatus] = useState<StatusWithMessage | undefined>(undefined);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { minusToleranceIndex, plusToleranceIndex } = useMemo(() => {
```

Comment:
Actually, you don't need these index. You can move the code in the useEffect below with the setInputValue (for the warningStatus) and remove the useMemo (also add toleranceRanges in the dependencies)